### PR TITLE
Gradle rewrite

### PR DESCRIPTION
**Pull request by PotatoPresident**
Rewrote Gradle build script in Kotlin DSL
Switched dependencies to use a version catalog
Add Github Actions workflow to deploy docs on release

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+        sourceCompatibility = javaVersion.toString()
+        targetCompatibility = javaVersion.toString()
+    }
+
+    jar {
+        from("LICENSE")
+    }
+
+    java {
+        toolchain { languageVersion.set(JavaLanguageVersion.of(javaVersion.toString())) }
```
Comment (yitzy299) [inline build.gradle.kts:134]: As the mixins file also has java version, we should either expose that for gradle to automatically handle or remove the java version option entirely, in favour of hardcoding

File: src/main/resources/fabric.mod.json [hunk 2]
```diff
@@ -1,34 +1,34 @@
...
-    "fabricloader": ">=0.11.3",
-    "fabric": ">=0.34.8",
-    "fabric-language-kotlin": ">=1.6.0+kotlin.1.5.0",
-    "minecraft": "1.17.x"
-  }
+    "schemaVersion": 1,
+    "id": "${id}",
+    "version": "${version}",
+    "name": "${name}",
+    "description": "Logs world events and allows for them to be rolled back",
```
Comment (yitzy299) [inline src/main/resources/fabric.mod.json:6]: Any reason why this isn't handled by gradle?

File: src/main/resources/fabric.mod.json [hunk 3]
```diff
@@ -1,34 +1,34 @@
...
+    "authors": [
+        "Potatoboy9999"
+    ],
+    "contact": {
+        "homepage": "https://github.com/QuiltServerTools",
```
Comment (yitzy299) [inline src/main/resources/fabric.mod.json:11]: Probably should lead to the wiki

**Review by yitzy299 — CHANGES_REQUESTED** [review]
Something else I forgot to mention - the fabric.mod.json spec includes custom values:

> Custom fields
A custom field can be provided in the JSON. It is a dictionary of string keys to JSON elements, and the contents of said JSON elements are entirely arbitrary. The Fabric loader will make no effort to read its contents, aside from a situation in which custom features in older schema versions become official in future schema versions, as a compatibility measure - those will be adequately documented.
It is recommended that the keys be namespaced with the ID of the mod relying on them (if such a mod exists) to prevent conflicts.

We should probably add a custom field to the `fabric.mod.json` file with the discord link, as it's currently hardcoded in the status command

File: src/main/resources/fabric.mod.json (hunk 2 above)
Comment (PotatoPresident) [reply src/main/resources/fabric.mod.json:6]: It is?

Comment (yitzy299) [reply src/main/resources/fabric.mod.json:6]: The description - should that not be a property?

**Review by yitzy299 — APPROVED** [review]

**Pull request merged**